exams/pearson/audit.py

```diff
@@ -0,0 +1,272 @@
+        if not all(getattr(settings, key) for key in REQUIRED_SETTINGS):
+            raise ImproperlyConfigured(
+                "One of EXAMS_AUDIT_ENCRYPTION_PUBLIC_KEY or "
+                "EXAMS_AUDIT_ENCRYPTION_FINGERPRINT is required but not set"
+            )
+
+        if self.fingerprint in self.get_fingerprints():
+            self.configured = True
+            return
+
+        print(self.public_key)
+        result = self.gpg.import_keys(self.public_key)
+
+        if result.counts['not_imported'] != 0:
+            raise ImproperlyConfigured('Error importing GPG keys for exam auditing: {}'.format(result.summary()))
+
+        if self.fingerprint not in result.fingerprints:
+            raise ImproperlyConfigured(
+                "Exam auditing GPG key successfully imported but {actual} "
+                "does not contain fingerprint: {expected}".format(
+                    actual=result.fingerprints,
+                    expected=self.fingerprint,
+                )
+            )
+
+        self.configured = True
+
+    def upload_encrypted_file(self, filename, file_type):
+        """
+        Encrypt the file locally and upload it
+
+        Args:
+            filename:
+                absolute path to the local file
+            file_type (str):
+                the type of file, either RESPONSE or REQUEST
+
+        Returns:
+            str:
+                absolute path to the local encrypted file
+        """
+        encrypted_filename = '{}.gpg'.format(filename)
+
+        log.debug('Encrypting file %s to %s', filename, encrypted_filename)
+
+        try:
+            with open(filename) as source_file:
+                self.gpg.encrypt(
+                    source_file,
+                    self.fingerprint,
+                    armor=False,
+                    symmetric=False,
+                    output=encrypted_filename
+                )
+
+            return self.store.upload(encrypted_filename, file_type)
+        finally:
+            self._cleanup(encrypted_filename)
+
+    def _cleanup(self, encrypted_filename):  # pylint: disable=missing-docstring
+        if os.path.exists(encrypted_filename):
+            try:
+                os.remove(encrypted_filename)
+            except:  # pylint: disable=bare-except
+                pass
+
+    def audit(self, filename, file_type):
+        """
+        Audits the given file
+        This means we encrypt it and store it on S3
+
+        Args:
+            filename (str):
+                path to the unencrypted file
+            file_type (str):
+                the type of file, either RESPONSE or REQUEST
+        Returns:
+            str:
+                the path where the file was stored
+        """
+        if not settings.EXAMS_AUDIT_ENABLED:
+            return None
+
+        self.configure()
+
+        return self.upload_encrypted_file(filename, file_type)
+
+    def audit_response(self, filename):
```

Comment:
nit: can you call these methods `audit_response_file` and `audit_request_file` respectively?